### PR TITLE
Add some specific debugging hint to email queue recipient empty

### PR DESCRIPTION
**Pull request by seansan**
No idea what works best. Ideally I would want to log ID/SUBJECT/STORE so I know where to start looking if this happens. If you propose otherwise, then please adjust the pull request directly and push. Only small win here is a little more specific debugging.

https://stackoverflow.com/questions/53581543/exception-mage-core-exception-with-message-message-recipients-data-must-be-se

### Fixed Issues (if relevant)

1. Fixes OpenMage/magento-lts#<issue_number>

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/core/Mage/Core/Model/Email/Queue.php [hunk 1]
```diff
@@ -96,7 +96,8 @@ protected function _afterSave()
     protected function _beforeSave()
     {
         if (empty($this->_recipients) || !is_array($this->_recipients)) {
-            Mage::throwException(Mage::helper('core')->__('Message recipients data must be set.'));
+            $error = Mage::helper('core')->__('Message recipients data must be set.');
+            Mage::throwException($error . " Msg Id:" . $this->getId());
```
Comment (kiatng) [inline app/code/core/Mage/Core/Model/Email/Queue.php:100]: Were you able to get the ID? I got a feeling that it's probably `null`.

**Comment by seansan** [comment]
In this case the ID and subject exist. The method checks for empty
recipients ... which is a good thing to do ... bt if it occurs you would
want to know "which" message entity has this problem

In this (rare) case we had 1 data row in queue but not corresponding data
in the linked recipients table

**Review by kiatng — DISMISSED** [review]

**Comment by fballiano** [comment]
I fixed conflicts and slightly changed the message so that doesn't create a problem with translations. I think this can be merged.

**Review by fballiano — APPROVED** [review]

**Review by ADDISON74 — APPROVED** [review]

**Comment by fballiano** [comment]
merged and cherrypicked to v20

**Pull request merged**